src/TestProblem/Hydro/CCSN/Record_CCSN.cpp

```diff
@@ -303,3 +346,103 @@ void Record_CCSN_GWSignal()
 
 } // FUNCTION : Record_CCSN_GWSignal()
 #endif // ifdef GRAVITY
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Detect_CoreBounce
+// Description :  Check whether the core bounce occurs
+//
+// Note        :  1. Invoked by Record_CCSN()
+//             :  2. Based on two criteria:
+//                   --> (a) The central density is larger than 2e14
+//                       (b) Any cells within 30km has entropy larger than 3
+//-------------------------------------------------------------------------------------------------------
+void Detect_CoreBounce()
+{
+
+// (1) criterion 1: central density is larger than 2e14
+   if ( CCSN_CentralDens < 2e14 )   return;
+
+
+// (2) criterion 2: any cells within 30km has entropy larger than 3
+   const double BoxCenter[3] = { amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2] };
```

Comment:
might be good to explain where is the BoxCenter in comments.

Reply:
especially when BoxCenter is not in the center.

Reply:
One quick question here. On the value of  30km, should we evaluate the distance relative to box center or the cell with highest density?